From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Thu, 19 Feb 2026 05:16:09 +0100
Subject: xdp-permissions: Use permissions ID to save and check the
 permissions

While this just the same at the moment, we are going to split the IDs
for the snap case, so we need to ensure that everywhere a permission is
involved we always refer to the permission ID

Origin: https://github.com/flatpak/xdg-desktop-portal/pull/1764
---
 src/xdp-permissions.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/xdp-permissions.c b/src/xdp-permissions.c
index 2c4d2f0..73fdde7 100644
--- a/src/xdp-permissions.c
+++ b/src/xdp-permissions.c
@@ -40,7 +40,7 @@ xdp_get_permissions_sync (XdpAppInfo *app_info,
   g_autoptr(GVariant) out_perms = NULL;
   g_autoptr(GVariant) out_data = NULL;
   g_autofree char **permissions = NULL;
-  const char *app_id;
+  const char *permissions_id;
 
   if (!xdp_dbus_impl_permission_store_call_lookup_sync (permission_store,
                                                         table,
@@ -55,10 +55,10 @@ xdp_get_permissions_sync (XdpAppInfo *app_info,
       return NULL;
     }
 
-  app_id = xdp_app_info_get_id (app_info);
-  if (!g_variant_lookup (out_perms, app_id, "^a&s", &permissions))
+  permissions_id = xdp_app_info_get_permissions_id (app_info);
+  if (!g_variant_lookup (out_perms, permissions_id, "^a&s", &permissions))
     {
-      g_debug ("No permissions stored for: %s %s, app %s", table, id, app_id);
+      g_debug ("No permissions stored for: %s %s, app %s", table, id, permissions_id);
 
       return NULL;
     }
@@ -133,7 +133,7 @@ xdp_set_permissions_sync (XdpAppInfo         *app_info,
                                                                 table,
                                                                 TRUE,
                                                                 id,
-                                                                xdp_app_info_get_id (app_info),
+                                                                xdp_app_info_get_permissions_id (app_info),
                                                                 permissions,
                                                                 NULL,
                                                                 &error))
